libs/fgviewer/src/DebugServer.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fgviewer/DebugServer.h>
+
+#include "ApiHandler.h"
+
+#include <CivetServer.h>
+
+#include <utils/FixedCapacityVector.h>
+#include <utils/Hash.h>
+#include <utils/Log.h>
+
+#include <tsl/robin_set.h>
+
+#include <sstream>
+#include <string>
+#include <string_view>
+
+namespace {
+std::string const BASE_URL = "libs/matdbg/web";
+} // anonymous
+
+namespace filament::fgviewer {
+
+using namespace utils;
+
+std::string_view const DebugServer::kSuccessHeader =
+        "HTTP/1.1 200 OK\r\nContent-Type: %s\r\n"
+        "Connection: close\r\n\r\n";
+
+std::string_view const DebugServer::kErrorHeader =
+        "HTTP/1.1 404 Not Found\r\nContent-Type: %s\r\n"
+        "Connection: close\r\n\r\n";
+
+class FileRequestHandler : public CivetHandler {
```

Comment:
I kind of wonder how much of this can be abstracted into another lib depended on by both matdbg and fgviewer. But duplicating code is ok too in this instance.

Reply:
yes, that could become a sister project eventually.

Reply:
I'd like to do this after the fgviewer works, since at that time it would be easier to see which parts could really be abstracted...?

Reply:
ack